### PR TITLE
add - auth and protected routes

### PR DESCRIPTION
**Pull request by Vishnu-Aithal**

### CODE REVIEW

File: src/auth-functions/authHandler.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import axios from "axios";
+
+export const signInHandler = async (email, password) => {
+    try {
+        const response = await axios.post("/api/auth/login", {
+            email: email,
+            password: password,
+        });
+        console.log(response);
```
Comment (foolhardy21) [inline src/auth-functions/authHandler.js:9]: you can remove this `console.log` if not required.

File: src/auth-functions/authHandler.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        if (response.status === 200) {
+            const token = response.data.encodedToken;
+            const userId = response.data.foundUser._id;
```
Comment (foolhardy21) [inline src/auth-functions/authHandler.js:12]: suggestion - you can de-structure response object in line5 to get these properties then and there itself.

**Review by foolhardy21 — APPROVED** [review]
code looks good. Suggestion - You can move the event handlers from JSX to separate functions.

File: src/auth-functions/authHandler.js (hunk 1 above)
Comment (MayankKumar10) [inline src/auth-functions/authHandler.js:9]: Suggestion: remove console log

Comment (Vishnu-Aithal) [reply src/auth-functions/authHandler.js]: thanks for the review .done

File: src/auth-functions/authHandler.js (hunk 2 above)
Comment (Vishnu-Aithal) [reply src/auth-functions/authHandler.js]: thanks for the review. done

File: src/auth-functions/authHandler.js (hunk 1 above)
Comment (Vishnu-Aithal) [reply src/auth-functions/authHandler.js]: thanks for the review . done

**Pull request merged**